src/frontend/src/app/registration/registration.component.html

```diff
@@ -0,0 +1,68 @@
+<br/>
+<div class="row justify-content-center">
+  <form [formGroup] = "registrationForm" (ngSubmit)="register()">
+
+    <div class="form-group">
+      <label for="inputFirstname">Firstname</label>
+      <input type="text" class="form-control w-50" id="inputFirstname"
+             placeholder="Enter firstname" [(ngModel)]="userDto.firstname" formControlName="firstname">
+      <div *ngIf="registrationForm.controls.firstname.invalid && (registrationForm.controls.firstname.touched || registrationForm.controls.firstname.dirty)">
```

Comment:
ten if musi się znajdować w kontrolerze czyli w komponencie, stwórz na to jakąś funkcję, a tu tylko ngIf=i nazwa funkcji... coś xD